distribution/kpkginstall/runtest.sh

```diff
@@ -5,34 +5,25 @@
 ARCH=$(uname -m)
 REBOOTCOUNT=${REBOOTCOUNT:-0}
 
-# Output version of the kernel contained in the specified kernel package
-# tarball.
-#
-# Args: tarball
-# Output: kernel version, nothing if not found
-# Status: zero if version is found, non-zero on failure
 function get_kpkg_ver()
 {
-  case "$KPKG" in
+  echo "Extracting kernel version from ${KPKG_URL}" | tee -a ${OUTPUTFILE}
+
+  case "${KPKG_URL}" in
     *.tar.gz)
-            tar tf "$1" | sed -ne '/^boot\/vmlinu[xz]-[1-9]/ {s/^[^-]*-//p;q}; $Q1'
+            KPKG=${KPKG_URL##*/}
```

Comment:
You need to add `declare` here to avoid creating a global variable, if you'd like that.
I.e.: `declare KPKG=${KPKG_URL##*/}`, or even better `declare -r KPKG=${KPKG_URL##*/}` for a read-only variable. Also, lowercase, since it's not global: `declare -r kpkg=${KPKG_URL##*/}`.

And then you'd probably want to remove passing the argument to this function in the code below, since you're using global state now (KPKG_URL and the name of the dnf repo).